pkg/collector/process.go

```diff
@@ -108,8 +109,11 @@ func (cp *CollectingProcess) Start() {
 }
 
 func (cp *CollectingProcess) Stop() {
-	klog.V(4).Info("stopping the collecting process")
 	close(cp.stopChan)
+	cp.closeAllClients()
```

Comment:
Looks like we are using this only for UDP.
Wondering if UDP server and client connections can be gracefully shutdown without this.
If not, we should only call this for UDP and skip for TCP as it is not used there.

Reply:
Went through its usage and found that we can remove the usage of `client.errChan`. Done. Thanks!